### PR TITLE
feat: Update key, delete key, mark as immutable

### PR DESCRIPTION
**Pull request by stanleyjones**
## Description

This PR allows the owner of a key in the key-value store to do the following things:

- Update the key to a new value
- Delete the key (and remove it from local state)
- Mark the key as immutable (but keep it in the local state)

## Related Issue

Closes https://github.com/liftedinit/roadmap/issues/66 (Update)
Closes https://github.com/liftedinit/roadmap/issues/67 (Delete)
Closes https://github.com/liftedinit/roadmap/issues/20 (Immutable)

## Testing

Manually tested and unit tests pass.

## Breaking Changes (if applicable)

None.

## Screenshots (if applicable)

**KEYS**

Screenshot 2023-02-03 at 10 19 51 AM

**MARK IMMUTABLE WARNING**

Screenshot 2023-02-03 at 10 20 14 AM

**MARKED**

Screenshot 2023-02-03 at 10 20 24 AM

**DELETE WARNING**

Screenshot 2023-02-03 at 10 20 36 AM

**DELETED**

Screenshot 2023-02-03 at 10 20 44 AM

**ADD A NEW KEY**

Screenshot 2023-02-03 at 10 21 13 AM

**UPDATE NEW KEY**

Screenshot 2023-02-03 at 10 21 29 AM

## Checklist:

- [x] I have read and followed the CONTRIBUTING guidelines for this project.
- [x] I have added or updated tests and they pass.
- [x] I have added or updated documentation and it is accurate.
- [x] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "@emotion/react": "^11.10.4",
     "@emotion/styled": "^11.10.4",
     "@fontsource/rubik": "^4.5.11",
-    "@liftedinit/many-js": "github:liftedinit/many-js",
+    "@liftedinit/many-js": "github:stanleyjones/many-js#kv-params",
```
Comment (stanleyjones) [inline package.json:10]: Change this back once https://github.com/liftedinit/many-js/pull/93 merges.

File: src/features/services/components/data-settings.tsx [hunk 2]
```diff
@@ -101,7 +113,7 @@ export function DataSettings() {
               <Th>Key</Th>
               <Th>Value</Th>
               <Th>Tag</Th>
-              <Th>Edit</Th>
+              <Th></Th>
```
Comment (stanleyjones) [inline src/features/services/components/data-settings.tsx:116]: Not just editing anymore.

File: src/features/services/components/data-settings.tsx [hunk 3]
```diff
@@ -118,13 +130,41 @@ export function DataSettings() {
...
                 <Td>
                   {item.modifiable && (
-                    <EditIcon
-                      onClick={() => {
-                        setKey(item.key);
-                        setValue(item.value);
-                        onOpen();
-                      }}
-                    />
+                    <ButtonGroup spacing={3}>
```
Comment (stanleyjones) [inline src/features/services/components/data-settings.tsx:133]: This whole group only shows if the key is modifiable by the current user.

File: src/features/services/queries.ts [hunk 4]
```diff
@@ -136,3 +134,32 @@ export function useQueryValues(keys: string[]) {
...
 }
+
+export function useTransferKey() {
+  // eslint-disable-next-line
+  const [_, network] = useNetworkContext();
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: async (key: string) =>
+      await network?.kvStore.transfer({ key, newOwner: "maiyg" }),
```
Comment (stanleyjones) [inline src/features/services/queries.ts:145]: Assigning the owner to `maiyg` makes it immutable, as that identity is reserved by Many to be no one.

**Review by stanleyjones — COMMENTED** [review]
Some hopefully helpful comments.

**Review by tysloan — APPROVED** [review]

**Pull request merged**